src/en/toonily/src/eu/kanade/tachiyomi/extension/en/toonily/Toonily.kt

```diff
@@ -17,63 +18,80 @@ class Toonily : Madara(
-        val request = super.searchMangaRequest(page, query, filters)
-
-        val queries = request.url.queryParameterNames
-            .filterNot { it == "s" }
-
-        val newUrl = "$baseUrl/${searchPage(page, query)}".toHttpUrl().newBuilder().apply {
-            queries.map { q ->
-                request.url.queryParameterValues(q).map {
-                    this.addQueryParameter(q, it)
-                }
-            }
-        }.build()
-
-        return request.newBuilder()
-            .url(newUrl)
-            .build()
+        return super.searchMangaRequest(
+            page,
+            query.replace(titleSpecialCharactersRegex, " ").trim(),
+            filters,
+        )
     }
 
     override fun genresRequest(): Request {
         return GET("$baseUrl/search/?post_type=wp-manga", headers)
     }
 
-    // The source customized the Madara theme and broke the filter.
-    override val filterNonMangaItems = false
-
-    override val useNewChapterEndpoint: Boolean = true
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val newManga = SManga.create().apply {
+            url = manga.url.replace("/webtoon/", "/$mangaSubString/")
+        }
+        return super.mangaDetailsRequest(newManga)
+    }
 
-    override fun searchMangaSelector() = "div.page-item-detail.manga"
+    override fun chapterListRequest(manga: SManga): Request {
+        return mangaDetailsRequest(manga)
+    }
 
     override fun parseChapterDate(date: String?): Long {
         val formattedDate = if (date?.contains("UP") == true) "today" else date
         return super.parseChapterDate(formattedDate)
     }
 
+    private fun hdCoverInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val url = request.url
+
+        return if (
+            url.pathSegments.firstOrNull() == "wp-content" &&
+            url.pathSegments.lastOrNull()?.contains(sdCoverRegex) == true
+        ) {
+            try {
+                val newUrl = url.newBuilder()
+                    .removePathSegment(url.pathSegments.lastIndex)
+                    .addPathSegment(
+                        sdCoverRegex.replace(
+                            url.pathSegments.last(),
+                            "$2",
+                        ),
+                    ).build()
+                val newRequest = request.newBuilder()
+                    .url(newUrl)
+                    .build()
+
+                chain.proceed(newRequest)
+                    .also { assert(it.isSuccessful) }
+            } catch (_: Throwable) {
+                chain.proceed(request)
+            }
+        } else {
+            chain.proceed(request)
+        }
+    }
+
     companion object {
         val titleSpecialCharactersRegex = "[^a-z0-9]+".toRegex()
-        val trailingHyphenRegex = "-+$".toRegex()
+        val sdCoverRegex = Regex("""(-\d+x\d+)(\.\w+)${'$'}""")
```

Comment:
First capture group is never used. Requires changing other code as well:
```suggestion
        val sdCoverRegex = Regex("""(?:-\d+x\d+)(\.\w+)${'$'}""")
```